File: youpinhui-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController",function($scope,$controller,$http,brandService){
  
	$controller('baseController',{$scope:$scope});
	
    $scope.findAllBrand=function(){
        brandService.findAllBrand().success(
            function(response){
                $scope.list=response;
            }		
        )
        
    }
   
  
   $scope.findPage=function(page,size){
        brandService.findPage(page,size).success(
                function(response){
                    $scope.list=response.rows;
                    $scope.paginationConf.totalItems=response.total;
                })
    }
    
    
  $scope.save=function(){
     var obj =null;
        if($scope.entity.id!=null){
            obj = brandService.update($scope.entity);
        }else{
            obj = brandService.add($scope.entity);
        }			  
        obj.success(
              function(response){
              if(response.success){
                  $scope.reloadList();
              }else{
                  alert(response.message);
              }
      })
  }	
  
  $scope.findOne=function(id){
     brandService.findOne(id).success(
              function(response){
                  $scope.entity=response;
              })
  }
 
 
  
  
  $scope.delete=function(){
      if($scope.selectIds.length<1){
          return;
      }
    brandService.delete($scope.selectIds).success(function(response){
        if(response){
            $scope.reloadList();
            $scope.selectIds=[];
        }else{
            alert(response.message);
        }
    })
      
  }

  $scope.search=function(page,size){
        brandService.search(page,size,$scope.searchEntity).success(
                function(response){
                    $scope.list=response.rows;
                    $scope.paginationConf.totalItems=response.total;
                })
    }
})

File: youpinhui-manager-web/src/main/java/com/youpinhui/manager/controller/BrandController.java
package com.youpinhui.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.youpinhui.entity.PageResult;
import com.youpinhui.entity.Result;
import com.youpinhui.pojo.TbBrand;
import com.youpinhui.sellergoods.service.BrandService;

@RestController
@RequestMapping("/brand")
public class BrandController {

	@Reference(timeout=10000)
	private BrandService brandService;

	@RequestMapping("/findAll")
	public List<TbBrand> findAll() {
		
		return brandService.findAll();
	}
	
	
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int size) {
		return brandService.findPage(page, size);
	}
	
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand) {
		
		try {
			brandService.add(brand);
			return new Result(true,"insert success");
		} catch (Exception e) {
			// TODO: handle exception
			e.printStackTrace();
			return new Result(false,"insert failed");
		}
	}
	
	@RequestMapping("/findOne")
	public TbBrand findOne(long id) {
		return brandService.findOne(id);
	}
	
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand brand) {
		
		try {
			brandService.update(brand);
			return new Result(true,"update success");
		} catch (Exception e) {
			// TODO: handle exception
			e.printStackTrace();
			return new Result(false,"update failed");
		}
	}
	
	
	@RequestMapping("/delete")
	public Result delete(Long [] ids) {
		
		try {
			brandService.delete(ids);
			return new Result(true,"delete success");
		} catch (Exception e) {
			// TODO: handle exception
			e.printStackTrace();
			return new Result(false,"delete failed");
		}
	}
	
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand brand, int page,int size) {
		
		return brandService.findPage(brand,page, size);
	}
	
	
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList() {
		
		return brandService.selectOptionList();
	}
	
	
	
}
